Added instructions to install on Arch

goddard-guryon opened the pull request:
Even though winapps is not officially supported on Arch, its installation is quite easy. You just need to do a few extra steps, which I have added at the bottom of the readme. This should get winapps to work on almost all Arch installations.

Commit:
Added instructions to install on Arch

Even though winapps is not officially supported on Arch, its installation is quite easy. You just need to do a few extra steps, which I have added at the bottom of the readme. This should get winapps to work on almost all Arch installations.

goddard-guryon committed:
Corrected syntax

E404NNF commented:
looks like `libguestfs` is in the official repos

goddard-guryon commented:
@E404NNF indeed it is. My bad, I'll fix it

Commit:
Update README.md

libguestfs is in official repos, no need to use AUR helper for it

Commit:
Merge branch 'main' of https://github.com/goddard-guryon/winapps into main

bayazidbh commented:
Not sure if just me, but if I did that libvirt.conf one with the single line, then virt-manager can't connect with LXC, QEMU, and QEMU user at all. I'm using Manjaro, and it might be something with my own install that changed things up. Still, if the VM was connected fine and it suddenly can't connect, then that's something to check. Deleting the line gets thing working again just fine for me.

goddard-guryon commented:
@bayazidbh I'm not sure what the error is, but I'd guess it has something to do with virt-manager permissions for QEMU (assuming that the user has installed QEMU as super user, adding that line lets virt-manager access QEMU as normal user without root access). But maybe permission structure is different in Manjaro. Either way, if it works without adding the line in libvirt.conf, then there's nothing wrong in using it like that.

goddard-guryon committed:
Update README.md

Probably permission structure is a bit different in Manjaro, due to which one of the steps is not required there. So I updated the file to make it clearer.

Mustang-ssc commented:
Thanks, finally I was able to set this up.

After installation I ran into this issue when trying to execute Excel:
`cp: cannot create regular file '/home/mustangssc/.local/bin/winapps': No such file or directory
  Configuring Excel...tee: /home/mustangssc/.local/bin/excel-o365: No such file or directory
chmod: cannot access '/home/mustangssc/.local/bin/excel-o365': No such file or directory
 Finished.
  Configuring Word...tee: /home/mustangssc/.local/bin/word-o365: No such file or directory
chmod: cannot access '/home/mustangssc/.local/bin/word-o365': No such file or directory`

I am not sure If I missed something, but here solution if anyone else faces it:
Go to your downloaded winapps repo and copy all content from folders winapps/apps/ & winapps/bin/ in ~/.local/bin. Create folder if non-existing.

Be sure to copy the content of those folders, not the folders themselves.

Cornul11 commented:
I've tried the normal installation guide and this specialized Arch installation guide, but my VM installation hangs at the windows logo before the installation process even begins. Has anyone encountered this before?

goddard-guryon commented:
> I've tried the normal installation guide and this specialized Arch installation guide, but my VM installation hangs at the windows logo before the installation process even begins. Has anyone encountered this before?

Not sure if it is directly related to winapps or QEMU in the first place. Have you tried re-downloading the ISO, or increasing the CPU/RAM allocated to the VM? From what you said, it looks like the windows installation itself is the source of trouble, not the VM

the-real-t30d0r commented:
> Thanks, finally I was able to set this up.
> 
> After installation I ran into this issue when trying to execute Excel: `cp: cannot create regular file '/home/mustangssc/.local/bin/winapps': No such file or directory Configuring Excel...tee: /home/mustangssc/.local/bin/excel-o365: No such file or directory chmod: cannot access '/home/mustangssc/.local/bin/excel-o365': No such file or directory Finished. Configuring Word...tee: /home/mustangssc/.local/bin/word-o365: No such file or directory chmod: cannot access '/home/mustangssc/.local/bin/word-o365': No such file or directory`
> 
> I am not sure If I missed something, but here solution if anyone else faces it: Go to your downloaded winapps repo and copy all content from folders winapps/apps/ & winapps/bin/ in ~/.local/bin. Create folder if non-existing.
> 
> Be sure to copy the content of those folders, not the folders themselves.


That was my fix for Manjaro Linux, thank you :) 👍🏼
